omisego-sdk/src/main/java/co/omisego/omisego/security/KeyManagerMarshmallow.kt

```diff
@@ -21,37 +20,35 @@ import javax.crypto.spec.GCMParameterSpec
 // IV or Initialization Vector used in encryption and decryption. It must be the same value and contain 12 characters.
 @RequiresApi(Build.VERSION_CODES.M)
 internal class KeyManagerMarshmallow(
-        keyHolder: KeyHolder,
-        private val iv: String = String(ByteArray(12))
+    keyHolder: KeyHolder,
+    iv: String = String(ByteArray(12))
 ) : KeyManager(keyHolder) {
-
-    override val encryptCipher: Cipher by lazy {
-        Cipher.getInstance(AES_MODE).also { initCipher(it, Cipher.ENCRYPT_MODE) }
-    }
-    override val decryptCipher: Cipher by lazy {
-        Cipher.getInstance(AES_MODE).also { initCipher(it, Cipher.DECRYPT_MODE) }
-    }
+    
+    override val encryptCipher: Cipher
+        get() = Cipher.getInstance(AES_MODE).also { initCipher(it, Cipher.ENCRYPT_MODE) }
+    override val decryptCipher: Cipher
+        get() = Cipher.getInstance(AES_MODE).also { initCipher(it, Cipher.DECRYPT_MODE) }
```

Comment:
I fail to understand how this can solve the bug you mentioned about `iv` not being saved if using a random values not specified by the dev 🤔 

Doing this will only decrease performances, since `iv` is basically final, using `get()` or `lazy` for ciphers shouldn't change anything.

Any other reason to revert to creating a new cipher for each call?

Reply:
Sorry if I let you misunderstand the problem. Here are the problem and reason.

If we internally random the `IV` value, we need to change it every time we encrypt the message. For example, 

```kotlin
val one = omgKeyManager.encrypt("One".toByteArray()) // iv = IV0000000001
val two = omgKeyManager.encrypt("Two".toByteArray()) // iv = IV0000000002
val three = omgKeyManager.encrypt("Three".toByteArray()) // iv = IV0000000003

omgKeyManager.decrypt(one.toByteArray()) // iv = IV0000000001
omgKeyManager.decrypt(two.toByteArray()) // iv = IV0000000002
omgKeyManager.decrypt(three.toByteArray()) // iv = IV0000000003
```

Each calls we need to random new `IV`, right? For decryption, we need to use the same `IV` that we used to encrypt. So we need to keep all the random `IV` somewhere? which is over complicated, since we're already using the `AndroidKeyStore` to encrypt/decrypt the data.

So I think we shouldn't random a new IV, we will use the same `IV` for every encryption and decryption. Unfortunately, we need to call `initCipher` every time we want to change the `IV`. There's no way to change it without re-initialization.

Reply:
> To avoid the Cipher class detect that we use the same IV, we need to call initCipher every time we want to encrypt or decrypt.

Ooh ok, I didn't know about that, re-using IV will cause an exception? If so then I finally understand.

Not sure about the possible security issues of re-using IV though, not at all my area of expertise 😄

Reply:
It's not security best practice, but I think it enough for now 😄

Reply:
👍